### PR TITLE
Add fluid support to DEFC

### PR DESCRIPTION
**Pull request by kuba6000**

### CODE REVIEW

File: src/main/java/emt/gthandler/common/loader/EMT_RecipeAdder.java [hunk 1]
```diff
@@ -21,6 +22,15 @@ public static void addFusionCraftingRecipe(ItemStack[] inputs, ItemStack output,
         sFusionCraftingRecipes.addRecipe(true, inputs, new ItemStack[]{output}, null, null, null, aDuration, aEUt, aTier);
     }
 
+    public static void addFusionCraftingRecipe(ItemStack[] inputs, FluidStack[] fluidinputs, ItemStack output, FluidStack[] fluidoutputs, int aDuration, int aEUt, int aTier) {
```
Comment (Connor-Colenso) [inline src/main/java/emt/gthandler/common/loader/EMT_RecipeAdder.java:25]: Why not ItemStack[] output?

File: src/main/java/emt/gthandler/common/tileentities/machines/multi/DE_Fusion_Crafter.java [hunk 2]
```diff
@@ -173,12 +174,12 @@ public GT_Recipe.GT_Recipe_Map getRecipeMap() {
...
     public boolean checkRecipe(ItemStack aStack) {
         ItemStack[] tInputs = getCompactedInputs();
-        //FluidStack[] tFluids = getCompactedFluids(); unused for now
+        FluidStack[] tFluids = getCompactedFluids();
 
         if (tInputs.length > 0) {
             long tVoltage = getMaxInputVoltage();
             byte tTier = (byte) Math.max(1, GT_Utility.getTier(tVoltage));
-            GT_Recipe tRecipe = EMT_RecipeAdder.sFusionCraftingRecipes.findRecipe(getBaseMetaTileEntity(), false, gregtech.api.enums.GT_Values.V[tTier], null, tInputs);
+            GT_Recipe tRecipe = EMT_RecipeAdder.sFusionCraftingRecipes.findRecipe(getBaseMetaTileEntity(), false, gregtech.api.enums.GT_Values.V[tTier], tFluids, tInputs);
```
Comment (Connor-Colenso) [inline src/main/java/emt/gthandler/common/tileentities/machines/multi/DE_Fusion_Crafter.java:182]: It's a bit cleaner to just import V and use it than this but up to you. Not really important.

**Review by Connor-Colenso — CHANGES_REQUESTED** [review]

File: src/main/java/emt/gthandler/common/tileentities/machines/multi/DE_Fusion_Crafter.java (hunk 2 above)
Comment (kuba6000) [reply src/main/java/emt/gthandler/common/tileentities/machines/multi/DE_Fusion_Crafter.java]: Done

File: src/main/java/emt/gthandler/common/loader/EMT_RecipeAdder.java (hunk 1 above)
Comment (kuba6000) [reply src/main/java/emt/gthandler/common/loader/EMT_RecipeAdder.java]: Ok I made 
- one generic with all arrays 
- one overload with multiple inputs, one output, no fluids
- one overload with multiple inputs, one fluid input, one output, one fluid output

should be enough i think

**Review by Connor-Colenso — APPROVED** [review]

**Pull request merged**